components/cmake/modules/FindMCT.cmake

```diff
@@ -0,0 +1,22 @@
+# - Try to find MCT
+#
+# MCT does not provide a cmake config file, so we have to write our
+# own find module.
+#
+# This can be controlled by setting the MCT_DIR (or, equivalently, the
+# MCT_ROOT environment variable)
+#
+# Once done, this will define:
+#
+#   The "mct" target
+#
+
+# Look for libmct in INSTALL_SHAREDPATH/lib
+find_library(MCT_LIB  mct  REQUIRED PATHS ${INSTALL_SHAREDPATH}/lib)
+find_library(MPEU_LIB mpeu REQUIRED PATHS ${INSTALL_SHAREDPATH}/lib $ENV{mct_ROOT})
+
+# Create the interface library, and set target properties
+add_library(mct INTERFACE)
+target_link_libraries(mct INTERFACE ${MCT_LIB} ${MPEU_LIB})
+target_include_directories(mct INTERFACE ${INSTALL_SHAREDPATH}/include)
+target_compile_definitions(mct INTERFACE "MCT_INTERFACE")
```

Comment:
Also not sure where this is use but there is code in CIME to allow MCT or MOAB to be used as the coupler.  We need to keep that.

Reply:
You mean we need to keep ability to use MOAB, not "keep MCT_INTERFACE in CPPDEFS", right?

Also, is there an example in e3sm of a compset using moab instead of mct?

Reply:
Yes that's what I mean.

Its not in master yet but you can run a moab case on this branch (on Chrysalis only for now) https://github.com/sarich/E3SM/tree/sarich/moab/update-moab-driver
e.g.  ./create_test SMS_Vmoab.ne4pg2_oQU480.WCYCL1850NS

Reply:
`MCT_INTERFACE` does not appear to be used in CIME either, but there's not much harm in keeping it.

Reply:
Imho, if something is unused (and pointless), it should be removed. Pointless code tends to pile up, until we have no idea of what is actually needed/used. I vote for pruning it.

Reply:
I think you are right.

Reply:
@rljacob , I looked at Sarich's Moab branch and he's not using this cppdef there either.